htdocs/core/modules/takepos/mod_takepos_ref_simple.php

```diff
@@ -133,10 +133,16 @@ public function getNextValue($objsoc = null, $invoice = null, $mode = 'next')
 
 		// First, we get the max value
 		$posindice = strlen($this->prefix.$pos_source.'-____-') + 1;
-		$sql  = "SELECT MAX(CAST(SUBSTRING(ref FROM ".$posindice.") AS SIGNED)) as max"; // This is standard SQL
-		$sql .= " FROM ".MAIN_DB_PREFIX."facture";
+		// $sql  = "SELECT MAX(CAST(SUBSTRING(ref FROM ".$posindice.") AS SIGNED)) as max"; // This is standard SQL
+		// $sql .= " FROM ".MAIN_DB_PREFIX."facture";
+		// $sql .= " WHERE ref LIKE '".$db->escape($this->prefix.$pos_source)."-____-%'";
+		// $sql .= " AND entity IN (".getEntity('invoicenumber', 1, $invoice).")";
+
+		$sql = "SELECT CAST(SUBSTRING(ref FROM ".$posindice.") AS SIGNED) as MAX ";
+		$sql .= " FROM ".MAIN_DB_PREFIX."facture ";
 		$sql .= " WHERE ref LIKE '".$db->escape($this->prefix.$pos_source)."-____-%'";
 		$sql .= " AND entity IN (".getEntity('invoicenumber', 1, $invoice).")";
+		$sql .= " ORDER BY rowid desc LIMIT 1;";
```

Comment:
Sorting on rowid instead of the number into ref, will return a value that is not necessarly the max, for example when some reocrd were introduced using import and rowid order does not match ref order.